File: java/Sorting.java
public class Sorting {

    public static void bubbleSort(int[] arr) {
        boolean swapped = true;
        int n = arr.length;

        while (swapped) {
            swapped = false;
            for (int i = 1; i < n; i++) {
                if (arr[i - 1] > arr[i]) {
                    int temp = arr[i];
                    arr[i] = arr[i - 1];
                    arr[i - 1] = temp;
                    swapped = true;
                }
            }
            n--;
        }
    }

    public static void printArray(int[] arr) {
        for (int v : arr) System.out.print(v + " ");
        System.out.println();
    }

    public static void main(String[] args) {
        int[] data = {5, 2, 9, 1, 3};

        System.out.println("Before sorting:");
        printArray(data);

        bubbleSort(data);

        System.out.println("After sorting:");
        printArray(data);
    }
}

File: cpp/sorting.cpp
#include <iostream>
#include <vector>
using namespace std;

void insertionSort(vector<int>& arr) {
    for (int i = 1; i < arr.size(); i++) {
        int key = arr[i];
        int j = i - 1;

        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key;
    }
}

void printArray(const vector<int>& arr) {
    for (int v : arr) cout << v << " ";
    cout << endl;
}

int main() {
    vector<int> data = {8, 4, 5, 2, 7};

    cout << "Before sorting:\n";
    printArray(data);

    insertionSort(data);

    cout << "After sorting:\n";
    printArray(data);

    return 0;
}
